site/faq.md

```diff
@@ -104,11 +170,127 @@ Now restart fluxd to re-read the k8s secret (if it is running):
+
+### How do I use a private git host (or one that's not github.com, gitlab.com, or bitbucket.com)?
+
+As part of using git+ssh securely from the Flux daemon, we make sure
+`StrictHostKeyChecking` is on in the
+[SSH config](http://man7.org/linux/man-pages/man5/ssh_config.5.html). This
+mitigates against man-in-the-middle attacks.
+
+We bake host keys for `github.com`, `gitlab.com`, and `bitbucket.com`
+into the image to cover some common cases. If you're using another
+service, or running your own git host, you need to supply your own
+host key(s).
+
+How to do this is documented in
+[setup.md](/site/standalone/setup.md#using-a-private-git-host).
+
+### Will Flux delete resources that are no longer in the git repository?
+
+Not at present. It's tricky to come up with a safe and unsurprising
+way for this to work. There's discussion of some possibilities in
+[weaveworks/flux#738](https://github.com/weaveworks/flux/issues/738).
+
+### Why does my CI pipeline keep getting triggered?
+
+There's a couple of reasons this can happen.
+
+The first is that Flux pushes commits to your git repo, and if that
+repo is configured to go through CI, usually those commits will
+trigger a build. You can avoid this by supplying the flag `--ci-skip`
+so that Flux's commit will append `[ci skip]` to its commit
+messages. Many CI system will treat that as meaning they should not
+run a build for that commit. You can use `--ci-skip-message`, if you
+need a different piece of text appened to commit messages.
+
+The other thing that can trigger CI is that Flux pushes a tag to the
+upstream git repo whenever it has applied new commits. This acts as a
+"high water mark" for Flux to know which commits have already been
+seen. The default name for this tag is `flux-sync`, but it can be
+changed with the flags `--git-sync-tag` and `--git-label`. The
+simplest way to avoid triggering builds is to exclude this tag from
+builds -- how to do that will depend on how your CI system is
+configured.
+
+Here's the relevant docs for some common CI systems:
+
+ - [CircleCI](https://circleci.com/docs/2.0/workflows/#git-tag-job-execution)
+ - [TravisCI](https://docs.travis-ci.com/user/customizing-the-build#Building-Specific-Branches)
+ - [GitLab](https://docs.gitlab.com/ee/ci/yaml/#only-and-except-simplified)
+
+## What is the "sync tag"; or, why do I see a `flux-sync` tag in my git repo?
+
+Flux keeps track of the last commit that it's applied to the cluster,
+by pushing a tag (controlled by the command-line flags
+`--git-sync-tag` and `--git-label`) to the git repository. This gives
+it a persistent high water mark, so even if it is restarted from
+scratch, it will be able to tell where it got to.
+
+Technically, it only needs this to be able to determine which image
+releases (including automated upgrades) it has applied, and that only
+matters if it has been asked to report those with the `--connect`
+flag. Future versions of Flux may be more sparing in use of the sync
+tag.
+
```

Comment:
You would need `—k8s-namespace-whitelist` set here or flux will fail to start when it can’t read some namespaces.

Reply:
Does this relate to the following section? I had not grasped that it fails utterly if you restrict the permissions, but don't also tell it to restrict its view.

Reply:
OK I tried this, by giving the flux service account these things:

 - a rolebinding to a do-anything role limited to a namespace
 - a clusterrolebinding to "get", "list" namespaces only

and flux responds to `list-controllers` with a permissions error. I'll add a note in the FAQ about using `--k8s-namespace-whitelist` (and maybe file a bug about doing better when we can't read things in a namespace).

Reply:
@justinbarrick If you get a chance, let me know if the fixup reads correctly -- ta!

Reply:
Looks good!